sdk/monitor/monitor-opentelemetry/src/metrics/quickpulse/utils.ts

```diff
@@ -20,23 +20,35 @@ import {
 } from "@opentelemetry/semantic-conventions";
 import { SDK_INFO, hrTimeToMilliseconds } from "@opentelemetry/core";
 import { DataPointType, Histogram, ResourceMetrics } from "@opentelemetry/sdk-metrics";
-import { AZURE_MONITOR_OPENTELEMETRY_VERSION } from "../../types";
+import {
+  AZURE_MONITOR_AUTO_ATTACH,
+  AZURE_MONITOR_OPENTELEMETRY_VERSION,
+  AZURE_MONITOR_PREFIX,
+} from "../../types";
 import { Resource } from "@opentelemetry/resources";
 import { QuickPulseMetricNames, QuickPulseOpenTelemetryMetricNames } from "./types";
+import { getOsPrefix } from "../../utils/common";
+import { getResourceProvider } from "../../utils/common";
 
+/** Get the internal SDK version */
 export function getSdkVersion(): string {
   const { node } = process.versions;
   const nodeVersion = node.split(".");
   const opentelemetryVersion = SDK_INFO[SemanticResourceAttributes.TELEMETRY_SDK_VERSION];
-
-  const prefix = process.env["AZURE_MONITOR_AGENT_PREFIX"]
-    ? process.env["AZURE_MONITOR_AGENT_PREFIX"]
-    : "";
   const version = `dst${AZURE_MONITOR_OPENTELEMETRY_VERSION}`;
-  const internalSdkVersion = `${prefix}node${nodeVersion}:otel${opentelemetryVersion}:${version}`;
+  const internalSdkVersion = `${process.env[AZURE_MONITOR_PREFIX]}node${nodeVersion}:otel${opentelemetryVersion}:${version}`;
```

Comment:
If AZURE_MONITOR_PREFIX is not available, is it possible this is adding 'undefined' as part of the string?, can you confirm?

Reply:
Updated this check. I also noticed that we split the node version so it prints something like `node20,11,0:otel1.21.0:dst1.2.0` with commas in the node version instead of `.`. Is that intended for version reporting?

Reply:
Well not according to spec right?, maybe a bug?

Reply:
Yeah, fix is up now.